src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -740,18 +740,14 @@ private static GcvScope getScope(
         Set<GcvScope> discoveredScopes = new HashSet<>(2);
         while (!stack.isEmpty()) {
             ResolvedDependencyResult dependent = stack.removeFirst();
-            if (dependent.isConstraint()
-                    || !(dependent.getRequested() instanceof ModuleComponentSelector)
-                    || traversedComponents.contains(dependent)) {
+            if (dependent.isConstraint() || !(dependent.getRequested() instanceof ModuleComponentSelector)) {
                 continue;
-            }
-
-            Optional<GcvScope> cachedValue = Optional.ofNullable(scopeCache.get(dependent.getFrom()));
-            if (cachedValue.isPresent()) {
-                discoveredScopes.add(cachedValue.get());
+            } else if (traversedComponents.contains(dependent)) {
                 continue;
             }
 
+            traversedComponents.add(dependent);
```

Comment:
I take it this was the bug, we could be continuing before we add this dependent to traversedComponents.